localstack/services/awslambda/resource_providers/aws_lambda_function.py

```diff
@@ -0,0 +1,510 @@
+                    NODEJS_CFN_RESPONSE_CONTENT,
+                )
+                save_file(os.path.join(tmp_dir, "cfn-response.js"), NODEJS_CFN_RESPONSE_CONTENT)
+            elif (
+                properties["Runtime"].lower().startswith("python")
+                and "import cfnresponse" in zip_file
+            ):
+                save_file(os.path.join(tmp_dir, "cfnresponse.py"), PYTHON_CFN_RESPONSE_CONTENT)
+
+            # create zip file
+            zip_file = create_zip_file(tmp_dir, get_content=True)
+            code["ZipFile"] = zip_file
+        finally:
+            rm_rf(tmp_dir)
+    if _include_arch and "Architectures" in properties:
+        code["Architectures"] = properties.get("Architectures")
+    return code
+
+
+class LambdaFunctionProvider(ResourceProvider[LambdaFunctionProperties]):
+
+    TYPE = "AWS::Lambda::Function"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[LambdaFunctionProperties],
+    ) -> ProgressEvent[LambdaFunctionProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/FunctionName
+
+        Required properties:
+          - Code
+          - Role
+
+        Create-only properties:
+          - /properties/FunctionName
+
+        Read-only properties:
+          - /properties/Arn
+          - /properties/SnapStartResponse
+          - /properties/SnapStartResponse/ApplyOn
+          - /properties/SnapStartResponse/OptimizationStatus
+
+        IAM permissions required:
+          - lambda:CreateFunction
+          - lambda:GetFunction
+          - lambda:PutFunctionConcurrency
+          - iam:PassRole
+          - s3:GetObject
+          - s3:GetObjectVersion
+          - ec2:DescribeSecurityGroups
+          - ec2:DescribeSubnets
+          - ec2:DescribeVpcs
+          - kms:Decrypt
+          - lambda:GetCodeSigningConfig
+          - lambda:GetFunctionCodeSigningConfig
+          - lambda:GetRuntimeManagementConfig
+          - lambda:PutRuntimeManagementConfig
+
+        """
+        model = request.desired_state
+        lambda_client = request.aws_client_factory.lambda_
+
+        if not request.custom_context.get(REPEATED_INVOCATION):
+            request.custom_context[REPEATED_INVOCATION] = True
+
+            name = model.get("FunctionName")
+            if not name:
+                name = util.generate_default_name(request.stack_name, request.logical_resource_id)
+                model["FunctionName"] = name
+
+            kwargs = util.select_attributes(
+                model,
+                [
+                    "Architectures",
+                    "Description",
+                    "FunctionName",
+                    "Handler",
+                    "ImageConfig",
+                    "PackageType",
+                    "Layers",
+                    "MemorySize",
```

Comment:
After selection timeout and MemorySize need to be converted to int if they aren't already